### DISCUSSION TITLE
Found a comfort clipboard config of vim mode in zed using system clipboard

### DISCUSSION THREAD

**Discussion by gloridifice** [zed-industries/zed#10801] [Show and tell]
My vim habits of clipboard is that just "x" in visual mode and "y" in any mode write content into clipboard and other any other actions don't write things into clipboard. Switch vim, I spent some times to find a solution of this habit and I found it. 
I hope this method can help others too. Here are steps.

1. Set this value to "on_yank" in settings.json, that make yanking content is written into system clipboard.
```json
  "vim": {
    "use_system_clipboard": "on_yank"
  }

```

2. Use "editor::Cut" and "editor::Paste" replace raw actions. "editor::Cut" and "Paste" use system clipboard instead of vim clipboard. If you want some action using vim clipboard, you can use "vim::Delete", "vim::Yank".
```json
  {
    "context": "Editor && vim_mode == normal && !VimWaiting && !menu",
    "bindings": {
      "p": "editor::Paste"
    }
  },
  {
    "context": "Editor && vim_mode == visual && !VimWaiting && !menu",
    "bindings": {
      "x": "editor::Cut"
    }
  }
```

That's all :D

**Comment by julyfun** [comment]
Good setting, but if I copy a message from web browser then it can't be pasted into zed using `p`

**Reply by debarkamondal** [reply]
are you sure? Mine works. Although I don't want it to work.

**Comment by gloridifice** [comment]
By the way, I found that the behaviour of `editor::Paste` is not same with `vim::Paste`, that's a little annoying, now i'm using the way below.

settings.json
```json
  "vim": {
    "use_system_clipboard": "always"
  },
```
keymap.json
```json
    "context": "Editor && vim_mode == normal && !VimWaiting && !menu",
    "bindings": {
      ...
      "d": "editor::Delete",
      "x": ["workspace::SendKeystrokes", "a backspace escape l"],
      "d d": "editor::DeleteLine",
      "s": ["workspace::SendKeystrokes", "a backspace"],
      "c i w": ["workspace::SendKeystrokes", "v i w delete i"]
    }
```
```json
    "context": "Editor && vim_mode == visual && !VimWaiting && !menu",
    "bindings": {
      // visual, visual line & visual block modes
      ...
      "x": "vim::VisualDelete",
      "d": "editor::Delete",
      "s": "editor::Delete"
    }
```
I replaced all `vim::Delete` behaviours by editor's deleting actions, so that any deleting action wouldn't write things into my clipboard.

**Comment by julyfun** [comment]
```json
[
  {
    "context": "Editor && (vim_mode == normal || vim_mode == visual) && !VimWaiting && !menu",
    "bindings": {
      "d": ["workspace::SendKeystrokes", "\" _ d"],
      "D": ["workspace::SendKeystrokes", "\" _ D"],
      "c": ["workspace::SendKeystrokes", "\" _ c"],
      "C": ["workspace::SendKeystrokes", "\" _ C"]
    }
  },
]
```

Works nicely

**Reply by gloridifice** [reply]
Pretty useful!
